tests/api/test_adgroup.py

```diff
@@ -0,0 +1,113 @@
+from nose.tools import assert_equal
+from flask import url_for, json
+from tests.base import BaseTestCase
+from tests.populate_database import parse_csv
+from collections import defaultdict
+
+
+class TestAdgroup(BaseTestCase):
```

Comment:
Good point. Will fill out the missing parts.

As you might have noticed that we've introduced the session scope for all the HTTP POST and PUT requests. If there is no exception captured there, all the changes in that session should be committed successfully. Will double check it's actually in the database in the tests. Be safe :)